pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -288,22 +313,23 @@ func (a *actuator) deployGardenlet(ctx context.Context, shootClient kubernetes.I
 		return err
 	}
 
-	// Apply gardenlet chart
 	return shootClient.ChartApplier().Apply(ctx, filepath.Join(charts.Path, "gardener", "gardenlet"), v1beta1constants.GardenNamespace, "gardenlet", kubernetes.Values(values))
 }
 
 func (a *actuator) deleteGardenlet(ctx context.Context, shootClient kubernetes.Interface, managedSeed *seedmanagementv1alpha1.ManagedSeed, shoot *gardencorev1beta1.Shoot) error {
-	// Decode gardenlet configuration
 	gardenletConfig, err := helper.DecodeGardenletConfiguration(&managedSeed.Spec.Gardenlet.Config, false)
 	if err != nil {
 		return err
 	}
 
-	// Ensure seed secrets are deleted
 	if err := a.ensureSeedSecretsDeleted(ctx, &gardenletConfig.SeedConfig.SeedTemplate.Spec, managedSeed); err != nil {
 		return err
 	}
 
+	if err := a.deleteSNIIngressNamespace(ctx, shootClient, *gardenletConfig.SNI.Ingress.Namespace); err != nil {
+		return fmt.Errorf("failed to delete SNI ingress namespace %s in shoot %s: %w", *gardenletConfig.SNI.Ingress.Namespace, kutil.ObjectName(shoot), err)
+	}
+
 	// Prepare gardenlet chart values
```

Comment:
Shouldn't we check `gardenletConfig.SNI.Ingress.Namespace` is non-nil before dereferencing it (see above)?

What is this namespace supposed to contain? Typically namespace deletion may take some time, e.g. deletion of the `garden` namespace takes sometimes > 5 minutes. Should we ensure it's deleted before the managed seed itself is deleted, like for the `garden` namespace?

Note: I plan to refactor the deletion code in the near future to get rid of the current requeue-through-error behavior. So I could also take care to wait for this namespace if you would rather not do it in this PR, just let me know if you think it's a good idea.